src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -506,9 +511,10 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
     CORINFO_InstructionSet isa = HWIntrinsicInfo::lookupIsa(intrinsic);
 
-    if ((isa == InstructionSet_Vector256) && !compExactlyDependsOn(InstructionSet_AVX))
+    if (((isa == InstructionSet_Vector256) && !compExactlyDependsOn(InstructionSet_AVX)) ||
+        (((isa == InstructionSet_Vector512) && !compExactlyDependsOn(InstructionSet_AVX512F))))
```

Comment:
This one should be checking for `BW` + `DQ`.

Maybe we should have a `IsBaselineVector512Isa` check that covers the minimum ISA requirement for Vector512. That makes it easier to change in the future if we decide the extra complexity of supporting a scenario where only `F` exists is worthwhile.

Reply:
Done